### PR TITLE
Feature/admin tests

### PR DESCRIPTION
**Pull request by PavloJedi**

### CODE REVIEW

File: back-end/controllers/QuestionController.js [hunk 1]
```diff
@@ -85,6 +86,15 @@ const deleteQuestion = async (req, res) => {
   }
 };
 
+const getTestByUrl = async (req, res) => {
```
Comment (andriydzhrvts) [inline back-end/controllers/QuestionController.js]: What if different levels have the same unit name?
We need to check by level and unit

Comment (PavloJedi) [reply back-end/controllers/QuestionController.js]: edited to get this by level and unit

File: front-end/src/services/questionService.js [hunk 2]
```diff
@@ -10,12 +10,27 @@ export const getQuestionsUnitsByLevel = async (level) => {
   return data;
 };
 
-export const getTopicDataByUrl = async (params) => {
-  const { data } = await API.get('/api/questions/url', params);
+export const getTopicDataByUrl = async (url) => {
```
Comment (andriydzhrvts) [inline front-end/src/services/questionService.js]: Return to previous approach

File: front-end/src/components/questions/TopicsBody/TopicsBody.jsx [hunk 3]
```diff
@@ -16,6 +20,11 @@ const TopicsBody = ({ topics, selectHandler, fullscreen }) => {
         <>
           <p className={styles.topic__title}>{item.topic}</p>
           <FontAwesomeIcon icon={faArrowRight} />
+          {isAuthenticated && currentUser.role === 'admin' && (
```
Comment (andriydzhrvts) [inline front-end/src/components/questions/TopicsBody/TopicsBody.jsx]: Use user roles constants

File: front-end/src/pages/Questions/Questions.jsx [hunk 4]
```diff
@@ -60,27 +69,42 @@ const Questions = () => {
 
   const { data, loading } = useFetchUnits(isEdit, searching, selectedLevel, 'question');
 
+  const saveTestBtn = isAuthenticated && currentUser.role === 'admin' && (
```
Comment (andriydzhrvts) [inline front-end/src/pages/Questions/Questions.jsx]: The same as ^

File: front-end/src/routes/AppRouter.jsx [hunk 5]
```diff
@@ -108,6 +108,28 @@ const RouterWrapper = () => {
               </PrivateRoute>
             }
           />
+          <Route
+            path='/app/questions/edit/:id'
+            element={
+              isAuthenticated &&
+              currentUser.role === 'admin' && (
```
Comment (andriydzhrvts) [inline front-end/src/routes/AppRouter.jsx]: The same

File: front-end/src/routes/AppRouter.jsx [hunk 6]
```diff
@@ -108,6 +108,28 @@ const RouterWrapper = () => {
...
+                  <Tests />
+                </PrivateRoute>
+              )
+            }
+          />
+          <Route
+            path='/app/questions/new'
+            element={
+              isAuthenticated &&
+              currentUser.role === 'admin' && (
```
Comment (andriydzhrvts) [inline front-end/src/routes/AppRouter.jsx]: Role constant

**Review by andriydzhrvts — APPROVED** [review]

**Pull request merged**